cryodrgn/commands_utils/write_cs.py

```diff
@@ -0,0 +1,64 @@
+'''
+Create a CryoSparc .cs file from a particle stack and ctf parameters, or an input .cs file
+'''
+
+import argparse
+import numpy as np
+import sys, os
+from cryodrgn import dataset
+from cryodrgn import utils
+log = utils.log 
+
+
+def add_args(parser):
+    parser.add_argument('particles', help='Input particles (.cs)')
+    parser.add_argument('--datadir', help='Data/Project directory for cryoSPARC')
+    parser.add_argument('--ctf', help='Input ctf.pkl')
+    parser.add_argument('--poses', help='Optionally include pose.pkl') 
+    parser.add_argument('--ind', help='Optionally filter by array of selected indices (.pkl)')
+    parser.add_argument('--full-path', action='store_true', help='Write the full path to particles (default: relative paths)')
+    parser.add_argument('-o', type=os.path.abspath, required=True, help='Output .star file')
+
+    return parser
+
+
+def main(args):
+    assert args.o.endswith('.cs'), "Output file must be .cs file"
+    input_ext = os.path.splitext(args.particles)[-1]
+    assert input_ext in ('.mrcs', '.txt', '.cs'), 'Input file must be .mrcs/.txt/.cs'
+
+    # Either accept an input cs file, or an input .mrcs/.txt with optional ctf/pose pkl file(s)
+    particles = dataset.load_particles(args.particles, lazy=False, datadir=args.datadir)
```

Comment:
I checked out this branch to try this new `write_cs` command, and it took forever to run and filled up the RAM. I had to kill it.
Shouldn't this line use `lazy=True`? Not sure I understand the full code path, but since this `write_cs` command is only generating a new .cs file that only contains metadata, I don't think it needs to read the actual particle images.

Reply:
Thanks @Guillawme - yes, you're right. Can you try checking out the `vb/issue150` branch again and give this a spin? Thanks much!

Reply:
Sorry it took me so long to test this. Now the command runs very quickly and I get a `.cs` file. 👍 

But when I try to import it in cryoSPARC (using the job type "Import result group"), the job crashes with an error saying "Result group to import must be located inside the project directory". I did copy the `.cs` file into the cryoSPARC project directory. Not sure how this is supposed to work, so I am probably doing something wrong.